File: teste-backEnd/Teste.App/viewModel/CrediarioViewModel.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text;
using Teste.Domain.Entities;

namespace Teste.App.viewModel
{
    [DataContract]
    public class CrediarioViewModel
    {
        [DataMember(Name = "id")]
        public int? Id { get; set; }

        [IgnoreDataMember]
        public Person Person { get; set; }

        [DataMember(Name = "person_id")]

        public int PersonId { get; set; }

        [DataMember(Name = "user_id")]
        public int UserId { get; set; }

        [IgnoreDataMember]
        public User User { get; set; }


        [DataMember(Name = "enable_crediario")]
        public Boolean EnableCrediario { get; set; }

        [DataMember(Name = "sales")]
        public virtual List<SaleViewModel> Sales { get; set; }
    }
}

File: frontEnd/src/app/viewModel/crediario.view-model.ts
import { SalesViewModel } from './sales.view-model';

export class CrediarioViewModel {
   id: number;
   person_id: number;
   user_id: number;
   sales: SalesViewModel[];
   enable_crediario: Boolean;
    constructor(i) {
        this.id = i.id != null ? i.id : null;
        this.person_id = i.person_id != null ? i.person_id : null;
        this.sales = i.sales != null ? i.sales : new SalesViewModel({});
        this.user_id = i.user_id != null ? i.user_id : null;
        this.enable_crediario = i.enable_crediario != null ? i.enable_crediario : null;
    }
}

